sphereon-kmp-crypto-kms-azure/src/jvmMain/kotlin/com/sphereon/crypto/kms/azure/AzureKeyVaultCryptoProvider.jvm.kt

```diff
@@ -0,0 +1,389 @@
+import com.azure.security.keyvault.certificates.CertificateClientBuilder
+import com.azure.security.keyvault.certificates.CertificateServiceVersion
+import com.azure.security.keyvault.keys.KeyAsyncClient
+import com.azure.security.keyvault.keys.KeyClientBuilder
+import com.azure.security.keyvault.keys.KeyServiceVersion
+import com.azure.security.keyvault.keys.models.CreateKeyOptions
+import com.azure.security.keyvault.keys.models.KeyCurveName
+import com.azure.security.keyvault.keys.models.KeyOperation
+import com.azure.security.keyvault.keys.models.KeyVaultKey
+import com.sphereon.crypto.CoseJoseKeyMappingService
+import com.sphereon.crypto.IKeyInfo
+import com.sphereon.crypto.SignClientException
+import com.sphereon.crypto.generic.CoseKeyPair
+import com.sphereon.crypto.generic.CryptoAlg
+import com.sphereon.crypto.generic.Curve
+import com.sphereon.crypto.generic.DigestAlg
+import com.sphereon.crypto.generic.JoseKeyPair
+import com.sphereon.crypto.generic.KeyOperations
+import com.sphereon.crypto.generic.KeyType
+import com.sphereon.crypto.generic.ManagedKeyPair
+import com.sphereon.crypto.generic.SignatureAlgorithm
+import com.sphereon.crypto.jose.JoseKeyOperations
+import com.sphereon.crypto.jose.JwaAlgorithm
+import com.sphereon.crypto.jose.JwaKeyType
+import com.sphereon.crypto.jose.Jwk
+import com.sphereon.crypto.jose.JwkUse
+import com.sphereon.crypto.jose.generateJwkThumbprint
+import com.sphereon.crypto.kms.IKeyManagementSystem
+import com.sphereon.crypto.sign.IRawSignatureService
+import com.sphereon.crypto.sign.ISimpleSignatureService
+import com.sphereon.crypto.sign.model.SignInput
+import com.sphereon.crypto.sign.model.SignOutput
+import com.sphereon.crypto.sign.model.Signature
+import com.sphereon.kmp.Logger
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.security.MessageDigest
+import java.time.Duration
+
+private val logger = Logger("sphereon:kmp:kms:azure-keyvault")
+
+actual class AzureKeyVaultCryptoProvider actual constructor(
+    id: String,
+    config: AzureKeyvaultClientConfig
+) : IKeyManagementSystem,
+    IRawSignatureService, ISimpleSignatureService {
+    private val id: String = "azure-keyvault"
+    private val keyClient: KeyAsyncClient
+    private val hasCerts: Boolean
+    private val isManagedHsm: Boolean
+    private val certClient: CertificateAsyncClient?
+
+    init {
+        keyClient = KeyClientBuilder()
+            .serviceVersion(KeyServiceVersion.V7_3)
+            .vaultUrl(config.keyvaultUrl)
+            .clientOptions(config.toClientOptions())
+            .retryPolicy(
+                if (config.exponentialBackoffRetryOpts == null) null
+                else RetryPolicy(RetryOptions(config.exponentialBackoffRetryOpts.toExponentialBackoffOptions()))
+            )
+            .credential(config.credentialOpts.toTokenCredential(config.tenantId))
+            .buildAsyncClient()
+
+        hasCerts = config.hsmType !== HSMType.MANAGED_HSM
```

Comment:
Why is this needed and what does it mean? Both HSMs can manage Certs, so the name is a bit confusing as I suspect it is meant for something else

Reply:
It was a left over from the code ported from eidas implementation. Removed.